README.md
The Yarn Gift Helper is designed to help non-knitters know how much yarn to gift the knitters in their life. 

With over 240,000 different yarns in the world*, many different thicknesses ("weights"), and yarn brands having different lengths of yarn in each ball/skein, it definately can be confusing.

## Knitting is...

"Knitting is at its fundamentals, a binary code featuring top-down design, standardized submodules, and recursive logic that relies on ratios, mathematical principles, and an intuitive grasp of three-dimensional geometry." ~ Kim Salazar KnitList 1995"

## The Developer as a Knitter:

I've been knitting since I was six years old, and typically have one sweater on the go plus smaller sock, mitten, hat or other projects. I greatly enjoy multi-colour stranded knitting ("Fairisle") and cables. I firmly believe that anyone can learn to knit - but not everyone has to learn. Find me on Ravely.com: https://www.ravelry.com/people/GingerKiwi

YarnHelp-Docs/CHANGELOG.md
*Note: This changelog includes both technical changes as well as the more personal process of, and reflections and influences on developing this app. Though the latter isn't typical of a commerical project, I've included it in this personal project to help others who are learning to build out their own personal projects.*

## 2.1 (2023.05.04 | May 04, 2023)

- create a dev branch in git for today's bit of work
- add all code for both knit and crochet double, queen, and king blankets
- delete "Comming Soon!" paragraph in double, queen, and king blankets
- merge dev branch to main
- Update CHANGELOG.md
- push to github

## 2.1 (2023.05.03 | May 03, 2023)

- create a dev branch in git for today's bit of work
- add all code for both knit and crochet throw and twin blankets
- delete "Comming Soon!" paragraph in twin blankets
- merge dev branch to main
- push to github
- ** Note this is definately not dry code! Way too much copy and paste.

## 2.1 (2023.03.29 | March 29, 2023)
March 29:
- Add crafting-and-code.html for the Crafting and Code Coffee Chats and discord.
- Update nav on all pages to include: 
- ``<li><a href="crafting-and-code.html">Crafting and Code Coffee Chats and Discord</a>``
after doctorwho.html.
- Add .margin-bottom-2rem to all li on 1st Crafting and Code section

## 2.1 (2023.03.27 | March 26-27, 2023)

March 27:
- Added eventListener back to howmuchyarn.html
- Added calculations for cat mats and baby blankets to blankets section on howmuchyarn.js
- Updated docs
- Tidied up comments in howmuchyarn.js
- Add image of tardis from Wikimedia Commons to doctorwho.html
- Update text on doctorwho.html 
	- Include ref to Chris Brimlow's Ravelry page and his DoctorWhoScarf.com website.
- Merge Dev-2023.03-add-crochet-and-blankets branch into main.
- Ready to push to GitHub and continuously deploy on Netflify.

March 26:
- Created doctorwho.html and doctorwho.js:
	- to debug what broke March 26
	- to have a spot to test pattern generator later
	- because it's cool and fun!
- Changed type="text" to type="number" for input on Marco Campos suggestion
- Found 2 bugs 
	- -> somehow deleted eventListener on March 26.
	- -> [Kieran](kieran.ca) on Toronto JS Slack help channel: answer-doctorwho-crochet-scarf-04 != answer-doctorwho-crochet-04 !
- Fixed the Doctor Who page

March 25: 
- Add scarfs to Calculations.md,  howmuchyarn.html,  and howmuchyarn01.js
- Fix bug that was causing a NaN error with help of Marco Campos on Discord video.
- Broke the site again.

## 2.0 (2023.03.18 | March 18, 2023)

### Work on Docs and Calculations
- Figured out that for **just the sweater calculations the app will need to do 200 calculations** for knit and crochet cardigans in 5 weights of yarn, for 6 sizes including oversized/long, cabled, and cabled cardigans.
	- Is it better to pull some of this data (eg total yardage per each project type/size) from a JSON doc?

- -> Goal is to do the calculations as variables outside of the string literals. The calculations are currently done inside the .innerHTML = statements like below:

```js
babyPulloverAnswer.innerHTML =

`You need ${Number(Math.ceil(550 * 1.05/yardsPerBall.value))} balls of this yarn for a baby pullover`
```

Instead something like this:
```js
// Worsted Weight (Craft Yarn Council yarn weight 4) for Baby Pullover:
const babyPulloverW4 = Number(Math.ceil(550 * 1.05/yardsPerBall.value))

babyPulloverAnswer.innerHTML = `${babyPulloverW4} balls of this yarn are needed for a baby pullover`
```

- Add PDF-References Folder to YarnHelp-Docs Folder
	- 2 webpages saved as pdf
		- Interweave Knits - sweater calculations
		- Twisted Yarn Shop - inc crochet, blankets/throws, scarfs, shawls, and accessories
	- Craft Yarn Council's Yarn Standards docs
- Add Calculations.md
	- Start list of yarn requirements for different weights of pullovers and cardigans
	- Start list of yarn requirements for blankets and throws
	- Add example of calculation in js code block

## 2.0 (2023.03.17 | March 17, 2023)

### Deploy to https://yarnhelp.app 

- Change domain name servers on Netlify to yarnhelp.app
- Deploy 1.0 site on https:yarnhelp.app
- Change [GitHub Repo](https://github.com/GingerKiwi/yarnhelp) name to yarnhelp
- Clone GitHub repo locally
- Create YarnHelp-Docs folder with CHANGELOG.md
- Create Obsidain vault for project to manage and compose markdown files
- Start actually using this changelog

## Start 2.0 Version

- Create Dev-2023.03 branch
- Create Yarnhelp1.0 branch (so I can easily show orginal version with live server by switching branches with/out rolling back)
- Rename orginal project files to:
	- index01.html
	- styles01.html
	- app01.html
- Create version 2.0 files for multipage site:
	- index.html
	- howmuchyarn.html
	- styles.css
	- howmuchyarn.css
	- howmuchyarn.js 
	- about.html
	- To show version 1.0 (building in public) include the following pages:
		- index01.html -> for version 1.0
			- Added link back to index.hmtl of ver 2.0
		- styles01.html
		- app01.js - > for version 1.0 

## 1.0.0 (2023.03.14 | March 14, 2023)

- Complete Scrimba's UI Design Course - which I partially started to upskill for yarnhelp.app
- Complete first half of Wes Bos' command line course. Realize how awesome the updates to my zsh and using iTerm2 are. 
- This is the push to finally get back to dev'ing yarnhelp.


## 1.0.0 (2023.01.00 | January, 2023)

### PWA Plans Start

- While skating on the Bentway Skate Trail have conversation with friend and senior dev Marco about technicalities of developing YarnHelp as a PWA.


## 1.0.0 (2022.10.05 | October 05, 2022)
Register yarnhelp.app on Google Domains

(Aside Note: also registered alicegame.app on the same day.)

## 1.0.0 (2022.09.22 | September 22, 2022)

Use metric convertor project design to create and deploy yarn-git-helper on Netlify at [yarngifthelper-gingerkiwi.netlify.app](https://yarngifthelper-gingerkiwi.netlify.app)

## 0.0.1 (2022-03-00 | March, 2022)


app01.js
let yardsPerBall = document.getElementById('yards-per-ball')
let calculateBtn = document.getElementById("btn-calculate")

let babyAnswer = document.getElementById('answer-baby')
let toddlerAnswer = document.getElementById('answer-toddler')
let childAnswer = document.getElementById('answer-child')
let adult1Answer = document.getElementById('answer-adult-1')
let adult2Answer = document.getElementById('answer-adult-2')

// -- HOW THE NUMBER OF BALLS NEEDED IS CALCULATED: -- //
// The number of balls needed is calculated using the "max yardage" in 
// https://www.interweave.com/article/knitting/how-much-yarn-do-i-need/" 
// adding a 5% buffer ( * 1.05) and rounding up to the whole ball/skein (Math.ceil)
// because yarn is sold by the ball/skein and not fraction)
// Max Yardage for Pullover Sweater/Jumper
// Baby: 550yards | Toddler: 800 yards | Child: 1200 yards | 
// Adult small/medium or Women's: 1400 yards | Adult large/extra large or Mens's: 1700 yards

calculateBtn.addEventListener("click", function() {

    
    // Baby Function //
    babyAnswer.innerHTML = 
        `You Need ${Number(Math.ceil(550 * 1.05/yardsPerBall.value))} Balls of Yarn`
    // Baby Function //
    toddlerAnswer.innerHTML = 
    `You Need ${Number(Math.ceil(800 * 1.05/yardsPerBall.value))} Balls of Yarn`
    // Baby Function //
    childAnswer.innerHTML = 
    `You Need ${Number(Math.ceil(1200 * 1.05/yardsPerBall.value))} Balls of Yarn`
    // Baby Function //
    adult1Answer.innerHTML = 
    `You Need ${Number(Math.ceil(1400 * 1.05/yardsPerBall.value))} Balls of Yarn`
    // Baby Function //
    adult2Answer.innerHTML = 
    `You Need ${Number(Math.ceil(1700* 1.05/yardsPerBall.value))} Balls of Yarn`
    
})


// babyAnswer.innerHTML = 
//         `For ${yardsPerBall.value} yards of yarn in a ball/skein you need ${Number(Math.ceil(550/yardsPerBall.value))} balls/skeins`

doctorwho.js
let yardsPerBall = document.getElementById('yards-per-ball')
let calculateBtn = document.getElementById("btn-calculate")

let DoctorWhoScarfAnswer04 = document.getElementById('answer-doctorwho-scarf-04')
let DoctorWhoScarfCrochetAnswer04 = document.getElementById('answer-doctorwho-crochet-scarf-04')

calculateBtn.addEventListener("click", function() {
    const yardsPerBallValue = Number.parseInt(yardsPerBall.value)

    const doctorWhoScarf04 = (Math.ceil(1260 * 1.05/yardsPerBallValue))
    const doctorWhoCrochetScarf04 = (Math.ceil(1260 * 1.28 * 1.05/yardsPerBallValue))

    DoctorWhoScarfAnswer04.innerHTML = `${doctorWhoScarf04} balls of this yarn are needed to knit a 12 foot long Doctor Who scarf.`
    DoctorWhoScarfCrochetAnswer04.innerHTML = `${doctorWhoCrochetScarf04} balls of this yarn are needed to crochet a 12 foot long Doctor Who scarf.`
})


howmuchyarn.js
let yardsPerBall = document.getElementById('yards-per-ball')
let calculateBtn = document.getElementById("btn-calculate")

let babyPulloverAnswer = document.getElementById('answer-pullover-baby')
let toddlerPulloverAnswer = document.getElementById('answer-pullover-toddler')
let childPulloverAnswer = document.getElementById('answer-pullover-child')
let adult1PulloverAnswer = document.getElementById('answer-pullover-adult01')
let adult2PulloverAnswer = document.getElementById('answer-pullover-adult02')


// -- NOTE: Need to refactor to use variables for each calculation
// See Calculations.md in YarnHelp-Docs for long form calculations and js code blocks to be put here.
// --- PULLOVER SWEATERS (JUMPERS) --- //
// Babies and Children //
// Number(Math.ceil(550 * 1.05/yardsPerBall.value))



calculateBtn.addEventListener("click", function() {

   // --- PULLOVER SWEATERS ---

    // Babies and Children //
    babyPulloverAnswer.innerHTML = 
        `You need ${Number(Math.ceil(550 * 1.05/yardsPerBall.value))} balls of this yarn for a baby pullover`
    toddlerPulloverAnswer.innerHTML = 
    `You need ${Number(Math.ceil(800 * 1.05/yardsPerBall.value))} balls of this yarn for a toddler pullover`
    childPulloverAnswer.innerHTML = 
    `You need ${Number(Math.ceil(1200 * 1.05/yardsPerBall.value))} balls of this yarn for a child`
    
    // Adults //
    adult1PulloverAnswer.innerHTML = 
    `You need ${Number(Math.ceil(1400 * 1.05/yardsPerBall.value))} balls of this yarn for a women's pullover`
    adult2PulloverAnswer.innerHTML = 
    `You need ${Number(Math.ceil(1700* 1.05/yardsPerBall.value))} balls of this yarn for a men's pullover`
    
    // --- BLANKETS AND THROWS

    // -- ACCESSORIES
})

// -- HOW THE NUMBER OF BALLS NEEDED IS CALCULATED - SWEATERS/JUMPERS: -- //
// The number of balls needed is calculated using the "max yardage" in 
// https://www.interweave.com/article/knitting/how-much-yarn-do-i-need/" 
// adding a 5% buffer ( * 1.05) and rounding up to the whole ball/skein (Math.ceil)
// because yarn is sold by the ball/skein and not fraction)
// Max Yardage for Pullover Sweater/Jumper
// Baby: 550yards | Toddler: 800 yards | Child: 1200 yards | 
// Adult small/medium or Women's: 1400 yards | Adult large/extra large or Mens's: 1700 yards

howmuchyarn01.js
// Note: the comments are more detailed than what I would put in production code for a commerical product so that new devs can learn from this project.
// Please see Calculations.md in the Docs folder on GitHub as well as the other .md and .pdf files to understand all things yarn and the calculations.
// The numbers refer to yarn weights (thicknesses). Please see docs especially the Craft Yarn Council yarn standards guide for info.

// If you're a new dev - Welcome to the community and have lots of fun! 
// Connect with me on LinkedIn: https://www.linkedin.com/in/elizabethmccready/

let yardsPerBall = document.getElementById('yards-per-ball')
let calculateBtn = document.getElementById('btn-calculate')

// =================//
// --- SWEATERS --- //
// =================//

// Knitting

let babyPulloverAnswer04 = document.getElementById('answer-pullover-baby-04')
let toddlerPulloverAnswer04 = document.getElementById('answer-pullover-toddler-04')
let childPulloverAnswer04 = document.getElementById('answer-pullover-child-04')
let adultSmallPulloverAnswer04 = document.getElementById('answer-pullover-adult-small-04')
let adultLargePulloverAnswer04 = document.getElementById('answer-pullover-adult-large-04')
let adultXxlPulloverAnswer04 = document.getElementById('answer-pullover-adult-xxl-04')

// Crochet

let babyCrochetPulloverAnswer04 = document.getElementById('answer-crochet-pullover-baby-04')
let toddlerCrochetPulloverAnswer04 = document.getElementById('answer-crochet-pullover-toddler-04')
let childCrochetPulloverAnswer04 = document.getElementById('answer-crochet-pullover-child-04')
let adultSmallCrochetPulloverAnswer04 = document.getElementById('answer-crochet-pullover-adult-small-04')
let adultLargeCrochetPulloverAnswer04 = document.getElementById('answer-crochet-pullover-adult-large-04')
let adultXxlCrochetPulloverAnswer04 = document.getElementById('answer-crochet-pullover-adult-xxl-04')

// ===============//
// --- SCARFS --- //
// ===============//

// Knitting

let childScarfAnswer04 = document.getElementById('answer-child-scarf-04')
let adultSmallScarfAnswer04 = document.getElementById('answer-adult-small-scarf-04')
let adultLargeScarfAnswer04 = document.getElementById('answer-adult-large-scarf-04')
let adultExtraLongScarfAnswer04= document.getElementById('answer-adult-extra-large-scarf-04')
let doctorWhoScarfAnswer04 = document.getElementById('answer-doctorwho-scarf-04')

// Crochet 

let childCrochetScarfAnswer04 = document.getElementById('answer-child-crochet-scarf-04')
let adultCrochetSmallScarfAnswer04 = document.getElementById('answer-adult-small-crochet-scarf-04')
let adultCrochetLargeScarfAnswer04 = document.getElementById('answer-adult-large-crochet-scarf-04')
let adultExtraLargeCrochetScarfAnswer04 = document.getElementById('answer-adult-extra-large-crochet-scarf-04')
let doctorWhoScarfCrochetAnswer04 = document.getElementById('answer-doctorwho-crochet-scarf-04')

// =================//
// --- BLANKETS --- //
// =================//

// Knitting

let catBlanketAnswer04 = document.getElementById('answer-cat-blanket-04')
let babyBlanketAnswer04 = document.getElementById('answer-baby-blanket-04')
let throwBlanketAnswer04 = document.getElementById('answer-throw-blanket-04')
let twinBlanketAnswer04 = document.getElementById('answer-twin-blanket-04')
let doubleBlanketAnswer04 = document.getElementById('answer-double-blanket-04')
let queenBlanketAnswer04 = document.getElementById('answer-queen-blanket-04')
let kingBlanketAnswer04 = document.getElementById('answer-king-blanket-04')


// Crochet 

let catCrochetBlanketAnswer04 = document.getElementById('answer-cat-crochet-blanket-04')
let babyCrochetBlanketAnswer04 = document.getElementById('answer-baby-crochet-blanket-04')
let throwCrochetBlanketAnswer04 = document.getElementById('answer-throw-crochet-blanket-04')
let twinCrochetBlanketAnswer04 = document.getElementById('answer-twin-crochet-blanket-04')
let doubleCrochetBlanketAnswer04 = document.getElementById('answer-double-crochet-blanket-04')
let queenCrochetBlanketAnswer04 = document.getElementById('answer-queen-crochet-blanket-04')
let kingCrochetBlanketAnswer04 = document.getElementById('answer-king-crochet-blanket-04')

// --- Make it all work! --- //

calculateBtn.addEventListener("click", function() {
    // Convert Input From User From String to Number
    const yardsPerBallValue = Number.parseInt(yardsPerBall.value)
    // --- 04 WORSTED WEIGHT (yarn thickness) CALCULATIONS --- //
    // --- 04 Pullovers --- //

    // 04 Knitting 
    const babyPullover04 = Number(Math.ceil(550 * 1.05/yardsPerBallValue))
    const toddlerPullover04 = Number(Math.ceil(800 * 1.05/yardsPerBallValue))
    const childPullover04 = Number(Math.ceil(1200 * 1.05/yardsPerBallValue))
    const adultSmallPullover04 = Number(Math.ceil(1400 * 1.05/yardsPerBallValue))
    const adultLargePullover04 = Number(Math.ceil(1700 * 1.05/yardsPerBallValue))
    const adultXxlPullover04 = Number(Math.ceil(1850 * 1.05/yardsPerBallValue))

    // 04 Crochet 
    const babyCrochetPullover04 = Number(Math.ceil(550 * 1.28 * 1.05/yardsPerBallValue))
    const toddlerCrochetPullover04 = Number(Math.ceil(800 * 1.28 * 1.05/yardsPerBallValue))
    const childCrochetPullover04 = Number(Math.ceil(1200 * 1.28 * 1.05/yardsPerBallValue))
    const adultSmallCrochetPullover04 = Number(Math.ceil(1400 * 1.28 * 1.05/yardsPerBallValue))
    const adultLargeCrochetPullover04 = Number(Math.ceil(1700 * 1.28 * 1.05/yardsPerBallValue))
    const adultXxlCrochetPullover04 = Number(Math.ceil(1850 * 1.28 * 1.05/yardsPerBallValue))

    // -- 04 Scarfs -- //
   
    // 04 Knitting 
    const childScarf04 = Number(Math.ceil(250 * 1.05/yardsPerBallValue))
    const adultSmallScarf04 = Number(Math.ceil(409 * 1.05/yardsPerBallValue))
    const adultLargeScarf04 = Number(Math.ceil(534 * 1.05/yardsPerBallValue))
    const adultExtraLongScarf04 = Number(Math.ceil(667 * 1.05/yardsPerBallValue))
    const doctorWhoScarf04 = Number(Math.ceil(1260 * 1.05/yardsPerBallValue))

    // 04 Crochet
    const childCrochetScarf04 = Number(Math.ceil(250 * 1.28 * 1.05/yardsPerBallValue))
    const adultCrochetSmallScarf04 = Number(Math.ceil(409 * 1.28 * 1.05/yardsPerBallValue))
    const adultCrochetLargeScarf04 = Number(Math.ceil(534 * 1.28 * 1.05/yardsPerBallValue))
    const adultExtraLargeCrochetScarf04 = Number(Math.ceil(667 * 1.28 * 1.05/yardsPerBallValue))
    const doctorWhoCrochetScarf04 = Number(Math.ceil(1260 * 1.28 * 1.05/yardsPerBallValue))

    // -- 04 Blankets -- //
   
    // 04 Knitting 

    const catBlanket04 = Number(Math.ceil(120 * 1.05/yardsPerBallValue))
    const babyBlanket04 = Number(Math.ceil(1340 * 1.05/yardsPerBallValue))
    const throwBlanket04 = Number(Math.ceil(2600 * 1.05/yardsPerBallValue))
    const twinBlanket04 = Number(Math.ceil(4950 * 1.05/yardsPerBallValue))
    const doubleBlanket04 = Number(Math.ceil(8100 * 1.05/yardsPerBallValue))
    const queenBlanket04 = Number(Math.ceil(8640 * 1.05/yardsPerBallValue))
    const kingBlanket04 = Number(Math.ceil(9720 * 1.05/yardsPerBallValue))


    // 04 Crochet
    const catCrochetBlanket04 = Number(Math.ceil(120 * 1.28 * 1.05/yardsPerBallValue))
    const babyCrochetBlanket04 = Number(Math.ceil(1340 * 1.28 * 1.05/yardsPerBallValue))
    const throwCrochetBlanket04 = Number(Math.ceil(2600 * 1.28 * 1.05/yardsPerBallValue))
    const twinCrochetBlanket04 = Number(Math.ceil(4950 * 1.28 * 1.05/yardsPerBallValue))
    const doubleCrochetBlanket04 = Number(Math.ceil(8100 * 1.28 * 1.05/yardsPerBallValue))
    const queenCrochetBlanket04 = Number(Math.ceil(8640 * 1.28 * 1.05/yardsPerBallValue))
    const kingCrochetBlanket04 = Number(Math.ceil(9720 * 1.28 * 1.05/yardsPerBallValue))

    // BLANKET YARDAGE REQUIREMENTS
        // cat 120
        // baby 1340
        // throw 2600
        // twin 4950
        // double 8100
        // queen 8640
        // king 9720

    // -- Use the consts above in Calculations to print answer statements to the <p> inside each categories answer.
    
    // 04 Pullover Answers
    babyPulloverAnswer04.innerHTML = `${babyPullover04} balls of this yarn are needed to knit a baby pullover.`
    babyCrochetPulloverAnswer04.innerHTML = `${babyCrochetPullover04} balls of this yarn are needed to crochet a baby pullover.`

    toddlerPulloverAnswer04.innerHTML = `${toddlerPullover04} balls of this yarn are needed to knit a toddler pullover.`
    toddlerCrochetPulloverAnswer04.innerHTML = `${toddlerCrochetPullover04} balls of this yarn are needed to crochet a toddler pullover.`

    childPulloverAnswer04.innerHTML = `${childPullover04} balls of this yarn are needed to knit a child pullover.`
    childCrochetPulloverAnswer04.innerHTML = `${childCrochetPullover04} balls of this yarn are needed to crochet a child pullover.`

    adultSmallPulloverAnswer04.innerHTML = `${adultSmallPullover04} balls of this yarn are needed to knit an adult small to medium size pullover.`
    adultSmallCrochetPulloverAnswer04.innerHTML = `${adultSmallCrochetPullover04} balls of this yarn are needed to crochet an adult small to medium size pullover.`

    adultLargePulloverAnswer04.innerHTML = `${adultLargePullover04} balls of this yarn are needed to knit an adult large to extra large adult pullover.`
    adultLargeCrochetPulloverAnswer04.innerHTML = `${adultLargeCrochetPullover04} balls of this yarn are needed to crochet an adult large to extra large adult pullover.`

    adultXxlPulloverAnswer04.innerHTML = `${adultXxlPullover04} balls of this yarn are needed to knit an adult oversized or cabled pullover.`
    adultXxlCrochetPulloverAnswer04.innerHTML = `${adultXxlCrochetPullover04} balls of this yarn are needed to crochet an adult oversized pullover.`

    // 04 Scarfs

    childScarfAnswer04.innerHTML = `${childScarf04} balls of this yarn are needed to knit a child scarf.`
    childCrochetScarfAnswer04.innerHTML = `${childCrochetScarf04} balls of this yarn are needed to crochet a child scarf.`

    adultSmallScarfAnswer04.innerHTML = `${adultSmallScarf04} balls of this yarn are needed to knit a small or medium adult scarf.`
    adultCrochetSmallScarfAnswer04.innerHTML = `${adultCrochetSmallScarf04} balls of this yarn are needed to crochet a  scarf.`

    adultLargeScarfAnswer04.innerHTML = `${adultLargeScarf04} balls of this yarn are needed to knit a large or extra large adult scarf.`
    adultCrochetLargeScarfAnswer04.innerHTML = `${adultCrochetLargeScarf04} balls of this yarn are needed to crochet a large or extra large adult scarf.`

    adultExtraLongScarfAnswer04.innerHTML = `${adultExtraLongScarf04} balls of this yarn are needed to knit an extra long scarf.`
    adultExtraLargeCrochetScarfAnswer04.innerHTML = `${adultExtraLargeCrochetScarf04} balls of this yarn are needed to crochet an extra long scarf.`

    doctorWhoScarfAnswer04.innerHTML = `${doctorWhoScarf04} balls of this yarn are needed to knit a 12 foot long Doctor Who scarf.`
    doctorWhoScarfCrochetAnswer04.innerHTML = `${doctorWhoCrochetScarf04} balls of this yarn are needed to crochet a 12 foot long Doctor Who scarf.`

    // 04 Blankets

    catBlanketAnswer04.innerHTML = `${catBlanket04} balls of this yarn are needed to knit a cat mat.`
    catCrochetBlanketAnswer04.innerHTML = `${catCrochetBlanket04} balls of this yarn are needed to crochet a cat mat.`

    babyBlanketAnswer04.innerHTML = `${babyBlanket04} balls of this yarn are needed to knit a baby blanket.`
    babyCrochetBlanketAnswer04.innerHTML = `${babyCrochetBlanket04} balls of this yarn are needed to crochet a baby blanket.`

    throwBlanketAnswer04.innerHTML = `${throwBlanket04} balls of this yarn are needed to knit a throw blanket.`
    throwCrochetBlanketAnswer04.innerHTML = `${throwCrochetBlanket04} balls of this yarn are needed to crochet a throw blanket.`

    twinBlanketAnswer04.innerHTML = `${twinBlanket04} balls of this yarn are needed to knit a twin blanket.`
    twinCrochetBlanketAnswer04.innerHTML = `${twinCrochetBlanket04} balls of this yarn are needed to crochet a twin blanket.`

    doubleBlanketAnswer04.innerHTML = `${doubleBlanket04} balls of this yarn are needed to knit a double blanket.`
    doubleCrochetBlanketAnswer04.innerHTML = `${doubleCrochetBlanket04} balls of this yarn are needed to crochet a double blanket.`

    queenBlanketAnswer04.innerHTML = `${queenBlanket04} balls of this yarn are needed to knit a queen blanket.`
    queenCrochetBlanketAnswer04.innerHTML = `${queenCrochetBlanket04} balls of this yarn are needed to crochet a queen blanket.`

    kingBlanketAnswer04.innerHTML = `${kingBlanket04} balls of this yarn are needed to knit a king blanket.`
    kingCrochetBlanketAnswer04.innerHTML = `${kingCrochetBlanket04} balls of this yarn are needed to crochet a king blanket.`

}) // New Devs: Dont' forget the last ) ! Nothing works if you forget it.


howmuchyarn-scarfs.js
let yardsPerBall = document.getElementById('yards-per-ball')
let calculateBtn = document.getElementById("btn-calculate")

const yardsPerBallValue = Number.parseInt(yardsPerBall.value);

// =========================//
// ---      SCARFS ---      //
// ========================= //

// Knitting Answers

// let babyPulloverAnswer04 = document.getElementById('answer-pullover-baby-04')
let ChildScarfAnswer04 = document.getElementById('answer-child-scarf-04')
let AdultSmallScarfAnswer04 = document.getElementById('answer-adult-small-scarf-04')
let AdultLargeScarfAnswer04 = document.getElementById('answer-adult-large-scarf-answer-04')
let AdultExtraScarfAnswer04 = document.getElementById('answer-adult-extra-scarf-answer-04')
let DoctorWhoScarfAnswer04 = document.getElementById('answer-doctorwho-scarf-answer-04')


// Crochet Answers

//let babyCrochetPulloverAnswer04 = document.getElementById('answer-crochet-pullover-baby-04')
let ChildCrochetScarfAnswer04 = document.getElementById('answer-child-crochet-scarf-04')
let AdultCrochetSmallScarfAnswer04 = document.getElementById('answer-adult-small-crochet-scarf-04')
let AdultCrochetLargeScarfAnswer04 = document.getElementById('answer-adult-large-crochet-scarf-04')
let AdultCrochetExtraScarfAnswer04 = document.getElementById('answer-adult-extra-crochet-scarf-answer-04')
let DoctorWhoScarfCrochetAnswer04 = document.getElementById('answer-doctorwho-crochet-scarf-04')

// --- 04 WORSTED WEIGHT (yarn thickness) CALCULATIONS --- //

// --- 04 Scarfs --- //
// SCARF YARDAGE REQUIREMENTS
// child 250
// adultSmall 409
// adultLarge 534
// adultExtra 667
// doctorWho 1260

// 04 Knitting 

const childScarf04 = Number(Math.ceil(250 * 1.05/yardsPerBallValue))
const adultSmallScarf04 = Number(Math.ceil(409 * 1.05/yardsPerBallValue))
const adultLargeScarf04 = Number(Math.ceil(534 * 1.05/yardsPerBallValue))
const adultExtraScarf04 = Number(Math.ceil(667 * 1.05/yardsPerBallValue))
const doctorWhoScarf04 = Number(Math.ceil(1260 * 1.05/yardsPerBallValue))

// 04 Crochet 
// Crochet takes approximately 25% more yarn than knitting. (* 1.25). An extra 3% has been added to 1.25 to make sure there is enouch yarn.

// const babyCrochetPullover04 = Number(Math.ceil(550 * 1.28 * 1.05/yardsPerBallValue))

const childCrochetScarf04 = Number(Math.ceil(250 * 1.28 * 1.05/yardsPerBallValue))
const adultCrochetSmallScarf04 = Number(Math.ceil(409 * 1.28 * 1.05/yardsPerBallValue))
const adultCrochetLargeScarf04 = Number(Math.ceil(534 * 1.28 * 1.05/yardsPerBallValue))
const adultCrochetExtraScarf04 = Number(Math.ceil(667 * 1.28 * 1.05/yardsPerBallValue))
const doctorWhoCrochetScarf04 = Number(Math.ceil(1260 * 1.28 * 1.05/yardsPerBallValue))

// Use the consts above in Calculations to print answer statements to the <p> inside each categories answer.

// babyPulloverAnswer04.innerHTML = `${babyPullover04} balls of this yarn are needed to knit a baby pullover`
// babyCrochetPulloverAnswer04.innerHTML = `${babyCrochetPullover04} balls of this yarn are needed to crochet a baby pullover`

ChildScarfAnswer04.innerHTML = `${childScarf04} balls of this yarn are needed to knit a child scarf`
ChildCrochetScarfAnswer04.innerHTML = `${childCrochetScarf04} balls of this yarn are needed to crochet a child scarf`

AdultLargeScarfAnswer04.innerHTML = `${adultSmallScarf04} balls of this yarn are needed to knit a small or medium adult scarf`
AdultCrochetSmallScarfAnswer04.innerHTML = `${adultCrochetSmallScarf04} balls of this yarn are needed to crochet a  scarf`

AdultLargeScarfAnswer04.innerHTML = `${adultLargeScarf04} balls of this yarn are needed to knit a large or extra large adult scarf`
AdultCrochetLargeScarfAnswer04.innerHTML = `${adultCrochetLargeScarf04} balls of this yarn are needed to crochet a large or extra large adult scarf`

adultExtraScarf04.innerHTML = `${adultExtraScarf04} balls of this yarn are needed to knit an extra long adult scarf`
adultCrochetExtraScarf04.innerHTML = `${adultCrochetExtraScarf04} balls of this yarn are needed to crochet an extra long adult scarf`

DoctorWhoScarfAnswer04.innerHTML = `${doctorWhoScarf04} balls of this yarn are needed to knit a child scarf`
DoctorWhoScarfCrochetAnswer04.innerHTML = `${doctorWhoCrochetScarf04} balls of this yarn are needed to crochet a child scarf`
